### DISCUSSION TITLE
QBX Support w ox

### DISCUSSION THREAD

**Discussion by khaosen** [alberttheprince/rpemotes-reborn#47] [Ideas]
Please keep in mind Qbox framework is not usable without ox.

Resource can use a Qbox support and to do that in most efficient way (without using qbus bridges) compatibility scripts would require ox. I am willing to pr the Qbox support made using ox library. Ox lines would only be used if framework is chosen as Qbox. Ox library would be disabled in fxmanifest.lua and would be manually undisabled shall the user select Qbox as framework.

![image](https://github.com/user-attachments/assets/3768dc01-3c52-40a2-adfc-cb2382163a9e)

**Comment by iSentrie** [comment]
Can we have a preview?

**Comment by khaosen** [comment]
I am waiting for approval to finish the Qbox support but basically qbox support scripts would use ox library, though, this script would only run if framework is selected as qbox.

**Comment by khaosen** [comment]
Also this would be a foundation for Qbox support. When Qbox adds their medical this will make it a lot easier to add qbox med support.

**Comment by alberttheprince** [comment]
I don't see an issue if this is added and is "opt in" as you're describing it, my only concern is if this would make updating the resource more time consuming in the future. I'm all for qbox/ox support/integration, I just want rpemotes to continue to be as broadly updated as possible, without having to update ox related things as well. 

If this can be done in a way that doesn't really affect updating that much, then I'm all for it.

Edit: Also worth noting that qbox officially lists [scully_emotemenu](https://github.com/Scullyy/scully_emotemenu) in their official recipe which uses ox_lib, but even before maintaining rpemotes-reborn, I always found rpemotes to be easier to use, have more content, easier to use, etc.

**Comment by khaosen** [comment]
I totally agree with you on the standalone matter but please keep in mind that Qbox as a framework relies on Ox. If we don't use Ox features then we will have to use Qbox's Qbux bridge connections which is literally same as the Qbus support rpemotes-reborn provides. As I said, Ox lines would only run if framework is selected as Qbox by the user. Since we also have to declare the ox in fxmanifest.lua, it would be disabled (--) as default and would be undisabled by user shall they select Qbox.
